packages/localize-tools/src/modes/transform.ts

```diff
@@ -331,9 +331,17 @@ class Transformer {
     if (templateResult.error) {
       throw new Error(stringifyDiagnostics([templateResult.error]));
     }
-    const {tag} = templateResult.result;
+    const {tag, contents} = templateResult.result;
     let {template} = templateResult.result;
 
+    const placeholdersByIndex = new Map<string, Placeholder>();
+    let idx = 0;
+    for (const content of contents) {
```

Comment:
I think here I'd probably do:

```ts
for (let i = 0; i < contents.length; i++) {
  const content = contents[i];
  ...
```

to keep the index variable more tightly scoped. I suppose you could also do:

```ts
{
  let idx = 0;
  for (const content of contents) {
    ...
}
```

but I find the first firm a more readable because it's a classic pattern, as opposed to the block which is a bit unusual

Reply:
I do want to keep `index` more tightly scoped but the former wouldn't work because it's not every `content` should increment, just the ones that are objects. So I think I'll go with the latter. Good call that curly braces can be used on their own for scoping!